src/k8-client/Cargo.toml

```diff
@@ -22,16 +22,11 @@ base64 = { version = "0.12.0" }
 futures-util = { version = "0.3.5"}
 rand = { version = "0.7.3" }
 isahc = { version = "0.9.9", features = ["json","static-curl"]}
-curl = { version = "0.4.33", features = ["mesalink"] }
```

Comment:
let's put `rustls`  related dependency into feature flag.  We still need to make that work.   Make sure `rustls` feature runs under CI as well

Reply:
current configuration works now so let's put all existing `rustls` configuration (minus working in hyper) in feature flag and make sure works in the CI